pkg/api/configsync/v1beta1/resource_override.go

```diff
@@ -73,6 +74,19 @@ type OverrideSpec struct {
 	// +listMapKey=containerName
 	// +optional
 	LogLevels []ContainerLogLevelOverride `json:"logLevels,omitempty"`
+
+	// namespaceStrategy controls how the reconciler handles Namespaces
+	// which are used by resources in the source but not declared.
+	// Must be "implicit" or "explicit"
+	// "implicit" means that the reconciler will implicitly create Namespaces
+	// if they do not exist, even if they are not declared in the source.
+	// "explicit" means that the reconciler will not create Namespaces which
+	// are not declared in the source.
+	//
+	// +kubebuilder:validation:Enum=implicit;explicit
+	// +optional
+	NamespaceStrategy configsync.NamespaceStrategy `json:"namespaceStrategy,omitempty"`
+	// TODO: For reviewers - should this be a RootSync-specific field?
```

Comment:
The `OverrideSpec` is currently shared between both RootSync and RepoSync. Should we make this a RootSync-only field?

WDYT @janetkuo ?

Reply:
Good point. Yes, this field should be RootSync only. Alternatively we could document this field as "it has no effect on RepoSync".

Reply:
SGTM. Do you have a preference?

Reply:
RootSync-only, imo

Reply:
Done. It's a RootSync-only field now. I split up OverrideSpec into a separate RootSync and RepoSync struct